--- /tmp/qlc-step3.c	2026-02-19 02:51:29.362154013 +0000
+++ bin/linux/obj/quickjs-libc.c	2026-02-19 02:51:22.544241240 +0000
@@ -167,6 +167,11 @@
 static uint64_t os_pending_signals;
 static int (*os_poll_func)(JSContext *ctx);
 
+void js_set_os_poll_func(int (*func)(JSContext *ctx))
+{
+    os_poll_func = func;
+}
+
 static void js_std_dbuf_init(JSContext *ctx, DynBuf *s)
 {
     dbuf_init2(s, JS_GetRuntime(ctx), (DynBufReallocFunc *)js_realloc_rt);
@@ -4048,7 +4053,8 @@
 
 static int js_os_init(JSContext *ctx, JSModuleDef *m)
 {
-    os_poll_func = js_os_poll;
+    if (!os_poll_func)
+        os_poll_func = js_os_poll;
 
 #ifdef USE_WORKER
     {
@@ -4346,7 +4352,7 @@
         }
 
         js_std_promise_rejection_check(ctx);
-        
+
         if (!os_poll_func || os_poll_func(ctx))
             break;
     }
